WarpX 23.07

ax3l opened the pull request:
Update WarpX and related Python packages to the lastest releases.

ax3l committed:
WarpX 23.07

Update WarpX and related Python packages to the lastest releases.

tldahlgren reviewed (dismissed):
Confirmed the new version sha256, rest LGTM.

tldahlgren commented:
Looks like the package fails to build for two CI stacks due to `py-picmistandard`.

adamjstewart commented:
@ax3l can you verify the checksums for picmicstandard and see where the incorrect ones are coming from? Might just be from GitHub instead of PyPI.

adamjstewart reviewed (dismissed):
Changes look good, just need to fix checksum

ax3l committed:
py-picmistandard: refresh hashes

Keeping even a single `git` version in here confuses the fetcher.
Remove broken old versions.

ax3l commented:
Checksums fixed! :tada: @tldahlgren @adamjstewart

tldahlgren reviewed (dismissed):
Confirmed all of the affected version sha256.  Thanks!

adamjstewart commented on var/spack/repos/builtin/packages/py-picmistandard/package.py:
```diff
@@ -11,27 +11,20 @@ class PyPicmistandard(PythonPackage):
-        sha256="b7eefdae1c43119984226b2df358c86fdeef7495084e47b3575e3d07e790ba30",
-        url="https://github.com/picmi-standard/picmi/archive/refs/tags/0.0.14.tar.gz",
-    )
-    version("0.0.14", sha256="8f83b25b281fc0309a0c4f75c7605afd5fa0ef4df3b3ac115069478c119bc8c3")
```
Technically we should deprecate these before removing them

ax3l replied:
Technically yes. But I think it's ok here - we really neither use nor support the first versions that had these install bugs and no proper tags.

Also, I do not have time to figure out the Spack fetcher bug, that decides to pick random git tags for some of the releases the moment I have a single version in here that was specified by git.

adamjstewart replied:
You're the maintainer, so the final decision is up to you 😄

ax3l committed:
Remove `py-warpx` versions with broken PICMI

adamjstewart commented on var/spack/repos/builtin/packages/py-picmistandard/package.py:
```diff
@@ -11,27 +11,20 @@ class PyPicmistandard(PythonPackage):
 
     maintainers("ax3l", "dpgrote", "RemiLehe")
 
     version("develop", branch="master")
```
```suggestion
    version("master", branch="master")
```

adamjstewart commented on var/spack/repos/builtin/packages/py-picmistandard/package.py:
```diff
@@ -11,27 +11,20 @@ class PyPicmistandard(PythonPackage):
-    )
-    version("0.0.14", sha256="8f83b25b281fc0309a0c4f75c7605afd5fa0ef4df3b3ac115069478c119bc8c3")
 
     depends_on("python@3.6:", type=("build", "run"))
```
You could remove this, Spack only supports 3.7+ anyway

ax3l replied:
Ok. Next release is 3.8+

adamjstewart commented on var/spack/repos/builtin/packages/py-warpx/package.py:
```diff
@@ -94,14 +78,13 @@ class PyWarpx(PythonPackage):
+    depends_on("py-picmistandard@0.25.0", type=("build", "run"), when="@23.07:")
     depends_on("py-setuptools@42:", type="build")
     # Since we use PYWARPX_LIB_DIR to pull binaries out of the
     # 'warpx' spack package, we don't need py-cmake as declared
```
Wheel is already added by the base class and could be removed here

adamjstewart commented on var/spack/repos/builtin/packages/py-warpx/package.py:
```diff
@@ -94,14 +78,13 @@ class PyWarpx(PythonPackage):
     depends_on("py-numpy@1.15.0:1", type=("build", "run"))
```
Python 3.7+ is the minimum version we support, so 3.6+ could be removed

adamjstewart commented on var/spack/repos/builtin/packages/py-warpx/package.py:
```diff
@@ -94,14 +78,13 @@ class PyWarpx(PythonPackage):
     depends_on("py-picmistandard@0.23.2", type=("build", "run"), when="@23.04:23.05")
-    depends_on("py-picmistandard@0.24.0", type=("build", "run"), when="@23.06:")
+    depends_on("py-picmistandard@0.24.0", type=("build", "run"), when="@23.06")
+    depends_on("py-picmistandard@0.25.0", type=("build", "run"), when="@23.07:")
```
I try to sort these from newest to oldest to match the version directive

adamjstewart reviewed (approved):
Some optional comments, feel free to merge if you think it's good enough

ax3l commented:
Thanks! I will merge this and add the suggestions to the 23.08 release :)
